packages/compass-indexes/src/modules/regular-indexes.ts

```diff
@@ -257,13 +304,59 @@ export const fetchIndexes = (
     }
   };
 };
-
 export const refreshRegularIndexes = (): IndexesThunkAction<
   Promise<void>,
   FetchIndexesActions
+> => {
+  return async (dispatch, getState) => {
+    const { status } = getState().regularIndexes;
+
+    // If we are in a READY state, then we have already fetched the data
+    // and are refreshing the list.
+    const reason: FetchReason =
+      status === FetchStatuses.READY
+        ? FetchReasons.REFRESH
+        : FetchReasons.INITIAL_FETCH;
+    await dispatch(fetchIndexes(reason));
+  };
+};
+
+export const pollRegularIndexes = (): IndexesThunkAction<
+  Promise<void>,
+  FetchIndexesActions
 > => {
   return async (dispatch) => {
-    return dispatch(fetchIndexes(true));
+    return await dispatch(fetchIndexes(FetchReasons.POLL));
+  };
+};
+
+export const POLLING_INTERVAL = 5000;
+
+let pollInterval: ReturnType<typeof setInterval> | undefined = undefined;
```

Comment:
Hmmm, I know that we're technically stop polling if the tab goes off screen, but I wonder if it would maybe make sense to already take the tab id into account somehow here. We have this clunky way of passing tab id around as a prop for the components, potentially we'd have to change that too if we'd decide to change that. Just thinking that if we ever allow to splitting tabs in the same window, this will cause an issue that's hard to spot

Reply:
Like `const pollIntervalByTabId = {};` ?

Reply:
We usually stick with `Map` for similar cases in other stores, but just an object like that would also work, yeah 👍

Reply:
Aah only saw this comment now. Changed to a Map.